rust/kernel/endian.rs

```diff
@@ -0,0 +1,162 @@
+// SPDX-License-Identifier: GPL-2.0
+
+//! Endian integer types
+
+macro_rules! define_le_integer {
+    ($name:ident, $native_type:ty) => {
+        /// little-endian integer
+        #[allow(non_camel_case_types)]
+        #[derive(Default, Clone, Copy)]
+        #[repr(transparent)]
+        pub struct $name($native_type);
+
+        impl From<$native_type> for $name {
+            #[cfg(target_endian = "little")]
+            fn from(v: $native_type) -> Self {
+                Self(v)
+            }
+            #[cfg(target_endian = "big")]
+            fn from(v: $native_type) -> Self {
+                Self(<$native_type>::from_le(v))
+            }
```

Comment:
Would `<$native_type>::to_le(v)` work on both little and big endian platforms?

Reply:
Yeah, it should work, with the overhead of one function call, I guess.